btclightclient/spv.go

```diff
@@ -0,0 +1,89 @@
+package btclightclient
+
+import (
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+)
+
+type SPVProof struct {
+	blockHash  chainhash.Hash
+	txId       string
+	txIndex    uint
+	merklePath []chainhash.Hash
+}
+
+type SPVStatus int
+
+const (
+	// proof is complete wrong
+	InvalidSPVProof SPVStatus = iota
+	// proof valid but the block have not "finalized" yet.
+	PartialValidSPVProof
+	// proof vlaid and block is "finalized".
+	ValidSPVProof
+)
+
+// / Get SPV proof from gettxoutproof Bitcoin API.
+func SPVProofFromHex(txoutProof string, txID string) (*SPVProof, error) {
+	// get block header
```

Comment:
```suggestion
```

Reply:
Rule 1. Comments should not duplicate the code.

https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments